src/plugins/data_source_management/public/components/data_source_selectable/data_source_selectable.tsx

```diff
@@ -81,67 +81,88 @@ export class DataSourceSelectable extends React.Component<
     this.setState({ ...this.state, isPopoverOpen: false });
   }
 
+  handleSelectedOption(dataSourceOptions: DataSourceOption[]) {
+    const [{ id }] = this.props.selectedOption!;
+    const dsOption = dataSourceOptions.find((ds) => ds.id === id);
+    if (!dsOption) {
+      this.props.notifications.addWarning(
+        i18n.translate('dataSource.fetchDataSourceError', {
+          defaultMessage: `Data source with id: ${id} is not available`,
+        })
+      );
+      this.setState({
+        ...this.state,
+        dataSourceOptions,
+        selectedOption: [],
+      });
+      this.props.onSelectedDataSources([]);
+      return;
+    }
+    this.setState({
+      ...this.state,
+      dataSourceOptions,
+      selectedOption: [{ id, label: dsOption.label, checked: 'on' }],
```

Comment:
@yujin-emma adding the prop here doesn't do anything, we should add to dataSourceOptions, can you check the previous implementation and add a test for this?